cmd/kube-controller-manager/app/plugins.go

```diff
@@ -69,6 +74,23 @@ func ProbeRecyclableVolumePlugins(flags VolumeConfigFlags) []volume.VolumePlugin
 	return allPlugins
 }
 
+// getCreatableVolumePluginForCloud maps a cloud provider to a specific volume plugin.
+func getCreatableVolumePluginForCloud(cloud cloudprovider.Interface, server *CMServer) volume.CreatableVolumePlugin {
+	var creater volume.VolumePlugin
+	switch {
+	case cloud == nil && server.EnableHostPathProvisioner:
```

Comment:
can this be simplified?  remove the hostpatch special case and you can make the switch cases much simpler.